c7n/resources/account.py

```diff
@@ -424,6 +425,86 @@ def process(self, resources, event=None):
+class SetAccountPasswordPolicy(BaseAction):
+    """Set an account's password policy.
+
+    This only changes the policy for the items provided.
+    If this is the first time setting a password policy and an item is not provided it will be
+    set to the defaults defined in the boto docs for IAM.Client.update_account_password_policy
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: set-account-password-policy
+                resource: account
+                filters:
+                  - not:
+                    - type: password-policy
+                      key: MinimumPasswordLength
+                      value: 10
+                      op: ge
+                actions:
+                    - type: set-password-policy
+                      policy:
+                        - key: MinimumPasswordLength
+                          value: 20
+    """
+    schema = type_schema(
+        'set-password-policy',
+        policy={
+            'type': 'array',
+            'items': {
+                'type': 'object',
+                'properties': {
+                    'key': {'type': 'string'},
+                    'value': {'oneOf': [{'type': 'integer'}, {'type': 'boolean'}]}
+                }
+            }
+        })
+    shape = 'PasswordPolicy'
+    permissions = ('iam:GetAccountPasswordPolicy', 'iam:UpdateAccountPasswordPolicy')
+    # We need to whitelist the values accepted by the update call as it is not available via shape
+    policy_whitelist = (
+        "MinimumPasswordLength", "RequireSymbols", "RequireNumbers", "RequireUppercaseCharacters",
+        "RequireLowercaseCharacters", "AllowUsersToChangePassword", "MaxPasswordAge",
+        "PasswordReusePrevention", "HardExpiry"
+    )
+
+    def validate(self):
+        policy = dict({item['key']: item['value'] for item in self.data['policy']})
+        attrs = policy
+        return shape_validate(
+            attrs,
+            self.shape,
+            'iam')
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('iam')
+        account = resources[0]
+        if not account.get('c7n:password_policy'):
+            try:
+                account['c7n:password_policy'] = client.get_account_password_policy().get(
+                    'PasswordPolicy', {})
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'NoSuchEntity':
+                    account['c7n:password_policy'] = {}
+                else:
+                    raise
+        for item in self.data.get('policy'):
+            account['c7n:password_policy'][item['key']] = item['value']
+        account['c7n:password_policy'] = {
+            k: v for (k, v) in account['c7n:password_policy'].items() if k in self.policy_whitelist
+        }
+        try:
+            client.update_account_password_policy(**account['c7n:password_policy'])
+        except Exception:
```

Comment:
blanket exceptions catches aren't kosher, and its not clear the purpose here since the handler does a raise, just omit it.

Reply:
Thanks for pointing this out @kapilt. Updated in b382acd.